plugin-build/src/main/kotlin/io/sentry/android/gradle/transforms/MetaInfStripTransform.kt

```diff
@@ -0,0 +1,122 @@
+        if (JarFile(input).isMultiRelease) {
+            // open a jar and do changes in-place
+            val uri = URI.create("jar:file:$input")
+            FileSystems.newFileSystem(uri, mapOf<String, Any>()).use { fs ->
+                val versionsDir = fs.getPath("META-INF/versions")
+                if (Files.exists(versionsDir)) {
+                    val unsupportedVersions = Files.list(versionsDir)
+                        .filter { (it.fileName.toString().toIntOrNull() ?: 0) > 11 }
+                        .toList()
+
+                    // unsupportedVersions contains paths like META-INF/versions/16
+                    // we walk the directory in a reverse order to delete all files/subdirectories
+                    // one-by-one and eventually delete the directory itself
+                    unsupportedVersions.forEach { path ->
+                        Files.walk(path)
+                            .sorted(Comparator.reverseOrder())
+                            .forEach { Files.delete(it) }
+                    }
+                }
+            }
+        }
+        outputs.file(input)
+    }
+
+    companion object {
+        internal val artifactType: Attribute<String> =
+            Attribute.of("artifactType", String::class.java)
+        internal val metaInfStripped: Attribute<Boolean> =
+            Attribute.of("meta-inf-stripped", Boolean::class.javaObjectType)
+
+        internal fun register(dependencies: DependencyHandler, forceInstrument: Boolean) {
+            // add our attribute to schema
+            dependencies.attributesSchema { schema ->
+                schema.attribute(metaInfStripped) { matchingStrategy ->
+                    /*
+                    * This is necessary so our transform is selected before the AGP's one
+                    * AGP's transform chain looks roughly like that:
+                    *
+                    * IdentityTransform (jar to processed-jar) -> IdentityTransform
+                    * (processed-jar to classes-jar) AsmClassesTransform (classes-jar to asm-transformed-classes-jar)
+                    *
+                    * We want out transform to run before the first IdentityTransform, so the chain
+                    * would look like that:
+                    *
+                    * MetaInfStripTransform (jar to processed-jar) -> IdentityTransform (jar to processed-jar)
+                    * -> ... -> ...
+                    *
+                    * Since the first two transforms have conflicting attributes, we define a
+                    * disambiguation rule to make Gradle select our transform first.
+                    */
+//                    matchingStrategy.disambiguationRules
+//                        .pickFirst { o1, o2 -> if (o1 > o2) -1 else if (o1 < o2) 1 else 0 }
+                }
+            }
+            // sets meta-inf-stripped attr to false for all .jar artifacts
+            dependencies.artifactTypes.named("jar") {
```

Comment:
What's about `aar`s?

Reply:
the multi-release capability is only applicable for jars I think. I couldn't find any info that you can create a mr-jar within an `aar`, so I guess I'd better leave it like this until someone requests it